packages/atomic/src/components/ipx/atomic-ipx-recs-list/atomic-recs-list/atomic-ipx-recs-list.tsx

```diff
@@ -0,0 +1,446 @@
+      );
+    }
+  }
+
+  private validateRecommendationIdentifier() {
+    const recListWithRecommendation = document.querySelectorAll(
+      `atomic-ipx-recs-list[recommendation="${this.recommendation}"]`
+    );
+
+    if (recListWithRecommendation.length > 1) {
+      this.bindings.engine.logger.warn(
+        `There are multiple atomic-ipx-recs-list in this page with the same recommendation property "${this.recommendation}". Make sure to set a different recommendation property for each.`
+      );
+    }
+  }
+
+  private updateOriginLevel2() {
+    if (this.label) {
+      const action = loadConfigurationActions(
+        this.bindings.engine
+      ).setOriginLevel2({
+        originLevel2: this.label,
+      });
+
+      this.bindings.engine.dispatch(action);
+    }
+  }
+
+  private renderHeading() {
+    if (!this.label) {
+      return;
+    }
+
+    if (this.recommendationListStateWithAugment.hasError) {
+      return;
+    }
+
+    if (
+      this.recommendationListStateWithAugment.firstRequestExecuted &&
+      !this.recommendationListStateWithAugment.hasItems
+    ) {
+      return;
+    }
+
+    return (
+      <Heading level={this.headingLevel} part="label" class="m-0 mb-2">
+        {this.bindings.i18n.t(this.label)}
+      </Heading>
+    );
+  }
+
+  private get currentIndex() {
+    return Math.abs(
+      (this.currentPage * this.numberOfRecommendationsPerPage!) %
+        this.recommendationListState.recommendations.length
+    );
+  }
+
+  private get subsetRecommendations() {
+    if (!this.numberOfRecommendationsPerPage) {
+      return this.recommendationListState.recommendations;
+    }
+
+    return this.recommendationListState.recommendations.slice(
+      this.currentIndex,
+      this.currentIndex + this.numberOfRecommendationsPerPage
+    );
+  }
+
+  private get numberOfPages() {
+    return Math.ceil(
+      this.recommendationListState.recommendations.length /
+        this.numberOfRecommendationsPerPage!
+    );
+  }
+
+  private get hasPagination() {
+    return !!this.numberOfRecommendationsPerPage;
+  }
+
+  private get shouldRenderPagination() {
+    return (
+      this.hasPagination && this.recommendationListStateWithAugment.hasItems
+    );
+  }
+
+  private async onSelect(
+    recommendation: RecsResult,
+    originalSelect: () => void
+  ) {
+    if (recommendation.raw.permanentid && this.actionsHistoryActions) {
+      const action =
+        this.actionsHistoryActions.addPageViewEntryInActionsHistory(
+          recommendation.raw.permanentid
+        );
+      this.bindings.engine.dispatch(action);
+    }
+    originalSelect();
+  }
+
+  private getPropsForAtomicRecsResult(recommendation: RecsResult) {
+    const interactiveResult = buildRecsInteractiveResult(this.bindings.engine, {
+      options: {result: recommendation},
+    });
+    const originalSelect = interactiveResult.select;
+    interactiveResult.select = () => {
+      this.onSelect(recommendation, originalSelect);
+    };
```

Comment:
This is the main change. We override the interactiveResult.onSelect method to call our custom onSelect method on line 325